driver/src/price_finding/linear_optimization_price_finder.rs

```diff
@@ -36,14 +36,27 @@ impl LinearOptimisationPriceFinder {
     }
 }
 
-fn token_id(token: usize) -> String {
+fn token_id(token: u16) -> String {
     format!("token{}", token)
 }
 
 fn account_id(account: H160) -> String {
     format!("{:x}", account)
 }
 
+fn serialize_tokens(orders: &[models::Order]) -> Vec<String> {
+    // Get collection of all token ids appearing in orders
+    let mut token_ids = orders.iter().map(|o| o.buy_token).collect::<HashSet<u16>>();
+    token_ids.extend(
+        orders
+            .iter()
+            .map(|o| o.sell_token)
+            .collect::<HashSet<u16>>(),
```

Comment:
nit: This extra collect is not needed, you should be able to call `extend` directly on an `Iterator`.